WordPress/Classes/ViewRelated/Blog/Site Picker/SitePickerViewController+SiteActions.swift

```diff
@@ -0,0 +1,101 @@
+import UIKit
+import SwiftUI
+
+extension SitePickerViewController {
+
+    func makeSiteActionsMenu() -> UIMenu? {
+        UIMenu(options: .displayInline, children: [
+            UIDeferredMenuElement.uncached { [weak self] in
+                $0(self?.makeSections() ?? [])
+            }
+        ])
+    }
+
+    private func makeSections() -> [UIMenu] {
+        [
+            makePrimarySection(),
+            makeSecondarySection(),
+            makeTertiarySection()
+        ]
+    }
+
+    private func makePrimarySection() -> UIMenu {
+        UIMenu(options: .displayInline, children: [
+            MenuItem.visitSite(visitSiteTapped),
+            MenuItem.switchSite(siteSwitcherTapped)
+        ].map { $0.toAction })
+    }
+
+    private func makeSecondarySection() -> UIMenu {
+        UIMenu(options: .displayInline, children: [
+            MenuItem.siteTitle(siteTitleTapped).toAction,
+            UIMenu(title: Strings.siteIcon, image: UIImage(systemName: "photo.circle"), children: [
+                makeSiteIconMenu() ?? UIMenu()
+            ])
+        ])
+    }
+
+    private func makeTertiarySection() -> UIMenu {
+        UIMenu(options: .displayInline, children: [
+            MenuItem.personalizeHome(personalizeHomeTapped).toAction
+        ])
+    }
+
+    private func personalizeHomeTapped() {
+        guard let siteID = blog.dotComID?.intValue else {
+            return DDLogError("Failed to show dashboard personalization screen: siteID is missing")
+        }
+
+        // TODO: track event
```

Comment:
Also need to track when the ellipsis button is tapped. I'll add these events in a separate PR

Reply:
I needed this recently too and learned `about `UIControlEventMenuActionTriggered`.